### PR TITLE
Parse cql type names

### PR DESCRIPTION
**Pull request by jorgebay**
Use `TEST_CASSANDRA_DIR` env variable to test against latest CASSANDRA-10365 branch. 

Handles UDT resolution.
Updated async dependency to ~1.5.

`SchemaParser` instances are not longer singletons.
Not tested with nested udt, waiting for dev branch to be updated.

### CODE REVIEW

File: lib/metadata/schema-parser.js [hunk 1]
```diff
@@ -698,29 +765,136 @@ SchemaParserV2.prototype._parseTableOrView = function (tableInfo, encoder, table
...
+        aggregate.returnType = type;
+        next(err);
+      });
+    }
+  ], function (err) {
+    if (err) {
+      return callback(err);
+    }
+    try {
+      aggregate.initCondition = encoder.decode(aggregate.initConditionRaw, aggregate.stateType);
```
Comment (tolbertam) [inline lib/metadata/schema-parser.js]: Will this work?  'initcond' is now a 'text' type, so i'm assuming this will be a string now instead of bytes.  Will debug a bit and see how it works.  I'm running into a variety of aggregate bugs around initcond in cassandra itself so hard to test at the moment.

File: lib/encoder.js [hunk 2]
```diff
@@ -52,7 +62,10 @@ var singleTypeNames = Object.freeze({
   'org.apache.cassandra.db.marshal.IntegerType':        dataTypes.varint,
   'org.apache.cassandra.db.marshal.CounterColumnType':  dataTypes.counter
 });
-var singleTypeNamesLength = Object.keys(singleTypeNames).reduce(function (previous, current) {
+var singleTypeNamesLength = Object.keys(dataTypes).reduce(function (previous, current) {
```
Comment (tolbertam) [inline lib/encoder.js]: This doesn't seem to be used from what i can tell

File: lib/encoder.js [hunk 3]
```diff
@@ -1021,15 +1034,176 @@ Encoder.prototype.handleBufferRef = function (buffer) {
   return buffer;
 };
 
+Encoder.prototype._parseUdtName = function (typeName, startIndex, length) {
```
Comment (tolbertam) [inline lib/encoder.js]: It looks like _parseUdtName also is defined with same code at line 1379

File: test/integration/short/metadata-tests.js [hunk 4]
```diff
@@ -165,7 +167,7 @@ describe('Metadata', function () {
             assert.ok(udt);
             assert.strictEqual(udt.fields.length, 2);
             assert.strictEqual(udt.fields[1].name, 'name');
-            assert.strictEqual(udt.fields[1].type.code, types.dataTypes.varchar);
+            assert.ok(udt.fields[1].type.code === types.dataTypes.varchar || udt.fields[1].type.code === types.dataTypes.text);
```
Comment (tolbertam) [inline test/integration/short/metadata-tests.js]: Interesting, why is this needed?  Does the value returned change from varchar to text now (I know they are aliases but was curious).

File: lib/metadata/schema-parser.js (hunk 1 above)
Comment (tolbertam) [inline lib/metadata/schema-parser.js]: It looks like it will not work as I suspected since `aggregate.initConditionRaw` is a string, get a TypeError at this line:

```
atolbert@atolbert-rmbp:~/Documents/Projects/nodejs-driver$ mocha test/integration/short/udf-tests.js -g 'should retrieve the most up to date metadata'


  Metadata
	...Starting test C* cluster v3.0.0 with 1 node(s)
	...With --install-dir=/Users/atolbert/Documents/Projects/cassandra
    #getFunction()
      ✓ should retrieve the most up to date metadata (5197ms)
    #getAggregate()
      1) should retrieve the most up to date metadata


  1 passing (17s)
  1 failing

  1) Metadata #getAggregate() should retrieve the most up to date metadata:
     Uncaught TypeError: bytes.readInt32BE is not a function
      at Encoder.decodeInt (lib/encoder.js:443:16)
      at Encoder.decode (lib/encoder.js:170:18)
      at lib/metadata/schema-parser.js:829:41
      at node_modules/async/lib/async.js:254:17
      at node_modules/async/lib/async.js:157:25
      at node_modules/async/lib/async.js:251:21
      at node_modules/async/lib/async.js:615:34
      at lib/metadata/schema-parser.js:821:9
      at Encoder.parseTypeName (lib/encoder.js:1157:12)
      at parseReturnType (lib/metadata/schema-parser.js:819:15)
      at node_modules/async/lib/async.js:610:21
      at node_modules/async/lib/async.js:249:17
      at iterate (node_modules/async/lib/async.js:149:13)
      at node_modules/async/lib/async.js:160:25
      at node_modules/async/lib/async.js:251:21
      at node_modules/async/lib/async.js:615:34
      at lib/metadata/schema-parser.js:815:9
      at Encoder.parseTypeName (lib/encoder.js:1157:12)
      at parseStateType (lib/metadata/schema-parser.js:813:15)
      at node_modules/async/lib/async.js:610:21
      at node_modules/async/lib/async.js:249:17
      at iterate (node_modules/async/lib/async.js:149:13)
      at node_modules/async/lib/async.js:160:25
      at node_modules/async/lib/async.js:251:21
      at node_modules/async/lib/async.js:615:34
      at lib/metadata/schema-parser.js:809:9
      at node_modules/async/lib/async.js:254:17
      at done (node_modules/async/lib/async.js:135:19)
      at node_modules/async/lib/async.js:32:16
      at node_modules/async/lib/async.js:251:21
      at Encoder.parseTypeName (lib/encoder.js:1157:12)
      at async.map.aggregate.argumentTypes (lib/metadata/schema-parser.js:806:17)
```

Comment (tolbertam) [inline lib/metadata/schema-parser.js]: Based on the comments in CASSANDRA-10365, i wonder if we just keep this stored as a text?  The java-driver has a way of parsing any type as a string into the types representation, but seems like a lot of effort to add here just for parsing this.

File: test/integration/short/metadata-tests.js (hunk 4 above)
Comment (jorgebay) [inline test/integration/short/metadata-tests.js]: Previously `varchar` and `text` were returned as Fqcn `UTF8Type`, now its either `text` or `varchar` depending on how it was defined.

File: lib/metadata/schema-parser.js (hunk 1 above)
Comment (jorgebay) [inline lib/metadata/schema-parser.js]: Makes sense to store it as an string, it was developed using a earlier version of CASSANDRA-10365 dev branch...

**Comment by jorgebay** [comment]
Updated branch based on @tolbertam feedback

**Comment by jorgebay** [comment]
`Aggregate.initCondition` is now a `string`.

File: test/integration/short/metadata-tests.js (hunk 4 above)
Comment (tolbertam) [inline test/integration/short/metadata-tests.js]: Ah cool that makes sense, thanks for the explanation :)
